adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter/ProtobufNoticeExporter.java

```diff
@@ -0,0 +1,303 @@
+                .build()
+                .writeTo(streamGenerator.getStream());
+    }
+
+    @Override
+    public void export(CannotParseFloatNotice toExport) throws IOException {
+        parsingNoticeToProto(toExport.getFilename(), toExport.getLineNumber(), toExport.getFieldName(),
+                toExport.getRawValue());
+    }
+
+    private void parsingNoticeToProto(String filename, int lineNumber, String fieldName, String rawValue) throws IOException {
+        protoBuilder.setCsvFileName(filename)
+                .setType(GtfsValidationOutputProto.GtfsProblem.Type.TYPE_CSV_VALUE_ERROR)
+                .setSeverity(GtfsValidationOutputProto.GtfsProblem.Severity.ERROR)
+                .setCsvColumnName(lineNumber, fieldName)
+                .setEntityValue(rawValue)
+                .build()
+                .writeTo(streamGenerator.getStream());
+    }
+
+    @Override
+    public void export(CannotParseIntegerNotice toExport) throws IOException {
+        parsingNoticeToProto(toExport.getFilename(), toExport.getLineNumber(), toExport.getFieldName(),
+                toExport.getRawValue());
+    }
+
+    @Override
+    public void export(CannotUnzipInputArchiveNotice toExport) throws IOException {
+        protoBuilder.setCsvFileName(toExport.getFilename())
+                .setType(GtfsValidationOutputProto.GtfsProblem.Type.TYPE_ARCHIVE_CORRUPTED)
+                .setSeverity(GtfsValidationOutputProto.GtfsProblem.Severity.ERROR)
+                .build()
+                .writeTo(streamGenerator.getStream());
+    }
+
+    @Override
+    public void export(FloatFieldValueOutOfRangeNotice toExport) throws IOException {
+        outOfRangeNoticeToProto(toExport.getFilename(), toExport.getEntityId(), toExport.getFieldName(),
+                String.valueOf(toExport.getRangeMin()), String.valueOf(toExport.getRangeMax()),
+                String.valueOf(toExport.getActualValue()));
+    }
+
+    @Override
+    public void export(IntegerFieldValueOutOfRangeNotice toExport) throws IOException {
+        outOfRangeNoticeToProto(toExport.getFilename(), toExport.getEntityId(), toExport.getFieldName(),
+                String.valueOf(toExport.getRangeMin()), String.valueOf(toExport.getRangeMax()),
+                String.valueOf(toExport.getActualValue()));
+    }
+
+    private void outOfRangeNoticeToProto(String filename, String entityId, String fieldName, String rangeMinAsString,
+                                         String rangeMaxAsString, String actualValueAsString)
+            throws IOException {
+        protoBuilder.setCsvFileName(filename)
+                .setType(GtfsValidationOutputProto.GtfsProblem.Type.TYPE_CSV_OUT_OF_RANGE)
+                .setSeverity(GtfsValidationOutputProto.GtfsProblem.Severity.ERROR)
+                .setAltEntityId(entityId)
+                .addCsvColumnName(fieldName)
+                .setValue(rangeMinAsString)
+                //TODO: FIXME - can't set altValue twice
```

Comment:
@lionel-nj Could you please fix that and the associated test?

Reply:
Sure, taking a look at this now.